docs/source/plugins/for_plugin_developers.rst

```diff
@@ -0,0 +1,291 @@
+<https://packaging.python.org/guides/creating-and-discovering-plugins/>`_.
+
+Using naming convention
+^^^^^^^^^^^^^^^^^^^^^^^
+
+``napari`` will look for *hook implementations* (i.e. functions decorated with
+the ``HookimplMarker("napari")`` decorator) in all top-level modules in
+``sys.path`` that begin with the name ``napari_`` (e.g. "``napari_myplugin``").
+
+One potential benefit of using discovery by naming convention is that it will
+allow ``napari`` to query the PyPi API to search for potential plugins.
+
+.. _entry-point-discovery:
+
+Using package metadata
+^^^^^^^^^^^^^^^^^^^^^^
+
+By providing an ``entry_points`` argument with the key ``napari.plugin`` to
+``setup()`` in ``setup.py``, plugins can register themselves for discovery
+(even if their names do not begin with "``napari_``").
+
+For example if you have a package named ``mypackage`` with a submodule
+``napari_plugin`` where you have decorated one or more napari hook
+implementations, then if you include in ``setup.py``:
+
+.. code-block:: python
+
+   # setup.py
+
+   setup(
+      ...
+      entry_points={'napari.plugin': 'plugin_name = mypackage.napari_plugin'},
+      ...
+   )
+
+... then napari will search the ``mypackage.napari_plugin`` module for
+functions decorated with the ``HookimplMarker("napari")`` decorator and
+register them the plugin name ``"plugin_name"``.
+
+One benefit of using this approach is that if you already have an existing
+pip-installable package, you can extend support for ``napari`` without having
+to rename your package, simply by identifying the module in your package that
+has the hook implementations.
+
+A user would then be able to use ``napari``, extended with your package's
+functionality by simply installing your package along with napari:
+
+.. code:: bash
+
+   pip install napari mypackage
+
+
+Cookiecutter template
+---------------------
+
+To quickly generate a new napari plugin project, you may wish to use the
+`cookiecutter-napari-plugin
+<https://github.com/napari/cookiecutter-napari-plugin>`_ template.  This uses
+the `cookiecutter <https://github.com/cookiecutter/cookiecutter>`_ command line
+utility, which will ask you a few questions about your project and get you
+started with a ready-to-go package layout where you can begin implementing your
+plugin.
+
+Install cookiecutter and use the template as follows:
+
+.. code-block:: bash
+
+   pip install cookiecutter
+   cookiecutter https://github.com/napari/cookiecutter-napari-plugin
+
+
+Example Plugins
+---------------
+
+For a minimal working plugin example, see the `napari-dv
+<https://github.com/tlambert03/napari-dv>`_ plugin, which allows ``napari`` to
+read the `Priism/MRC/Deltavision image file format
+<https://github.com/tlambert03/mrc>`_.
+
+For a more thorough plugin see `napari-aicsimageio
+<https://github.com/AllenCellModeling/napari-aicsimageio>`_, one of the first
+third-party plugins developed for napari.  This plugin takes advantage of
```

Comment:
```suggestion
community plugins developed for napari.  This plugin takes advantage of
```

I prefer community over third-party, curious what others think. Feels more inclusive

Reply:
Yeah, I started there actually... I don't know why but somehow it felt like it was "spreading credit" for Jackson's work lol... but I also prefer the word community